go/store/nbs/store.go

```diff
@@ -1625,9 +1625,9 @@ func (nbs *NomsBlockStore) swapTables(ctx context.Context, specs []tableSpec) (e
 	if err != nil {
 		return err
 	}
+	oldTables := nbs.tables
 	nbs.tables, nbs.upstream = ts, upstream
-
-	return nil
+	return oldTables.close()
```

Comment:
This was a leak on GC. Not relevant in the doltremoteapi context, but needs to be fixed for paranoid mode.